packages/babel-preset-env/src/index.ts

```diff
@@ -263,19 +268,23 @@ function getLocalTargets(
   });
 }
 
-function supportsStaticESM(caller) {
+function supportsStaticESM(caller: CallerMetadata | void) {
+  // @ts-expect-error optional chaining
   return !!caller?.supportsStaticESM;
 }
 
-function supportsDynamicImport(caller) {
+function supportsDynamicImport(caller: CallerMetadata | void) {
+  // @ts-expect-error optional chaining
   return !!caller?.supportsDynamicImport;
```

Comment:
Q: Why does TS error on optional chaining?

Reply:
Looks to be because of the `| void` type, should probably use `| undefined` instead.

Reply:
TS will throw can not find `supportsDynamicImport` in `void`, however it works with `CallerMetadata | undefined`. I  can change the caller extractor typings from `CallerMetadata | void` to `CallerMetadata | undefined` in another PR. However we did not define `supportsDynamicImport` in `CallerMetadata` so it will still throw.